docs/nix-develop.md

```diff
@@ -0,0 +1,206 @@
+# Using Nix Develop Environment
+
```

Comment:
Maybe we can detail here why we are using nix to build/develop, a brief summary with the advantages of nix?